### PR TITLE
Use cached logos and fallback icons for breach email

### PR DESCRIPTION
**Pull request by flozia**
# References: 
Jira: [MNTOR-1496](https://mozilla-hub.atlassian.net/browse/MNTOR-1496)

# Description

This adds the logos and fallback icon we introduced for the breaches to the email breach notifications.

# Screenshot

| Logo available | Logo fallback |
| --- | --- |
|  breach-notification-icon | breach-notification-fallback |

# How to test

Visit `/admin/emails/notification` and send yourself a test email.

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/utils/breach-logo.js [hunk 1]
```diff
@@ -5,30 +5,63 @@
...
-    '--blue-5',
-    '--purple-5',
-    '--green-05',
-    '--violet-5',
-    '--orange-5',
-    '--yellow-5',
-    '--red-5',
-    '--pink-5'
+    {
+      className: 'bg-blue-5',
```
Comment (flozia) [inline src/utils/breach-logo.js:34]: [CSS Variables in emails](https://www.caniemail.com/features/css-variables/) is lacking support.

Comment (Vinnl) [reply src/utils/breach-logo.js:34]: Ha, I'd be surprised if it had been the other way around :joy:

File: src/utils/breach-logo.js [hunk 2]
```diff
@@ -5,30 +5,63 @@
 /**
  * @param {object} breach
  * @param {Map<string, string>} logos Map of URLs to logos indexed by the domain name of the respective company
+ * @param {boolean} isEmail Is the icon being used in an email template?
```
Comment (Vinnl) [inline src/utils/breach-logo.js:8]: Suggestion, non-blocking: I think I'd do away with the conditionals, and just add the colour class-names regardless (with a comment that they're added for the emails), to simplify the code.

If you do want to keep `isEmail`, then I would at least recommend (like I'd do everywhere) to make all optional parameters named properties of an `options` parameter. That way, if we add more optional parameters in the future, we won't have to do `getBreachLogo(breach, logos, false, true, true, false)`, and it's easier to tell what the option does at the call site as well.

In other words:

```diff
- export function getBreachLogo (breach, logos, isEmail = false) {
+ export function getBreachLogo (breach, logos, options = {}) {
+   const isEmail = options.isEmail ?? false;
```

**Review by Vinnl — APPROVED** [review]
Code mostly looks good, but I don't think the embedded image works now in actual emails - though it's hard to verify when running locally.

File: src/views/emails/email-breach-card.js [hunk 3]
```diff
@@ -64,15 +71,15 @@ const breachCardPartial = breachData => {
...
-                  width='25'
-                >
-                ${Title}
-              </td>
+            <td style='${breachAlertCardsTitleStyle}'>
+              <span
+                class='breachLogoWrapper'
+                style='${breachAlertCardsTitleImageStyle}'
+              >
+                ${getBreachLogo(breachData, breachLogos, true)}
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/views/emails/email-breach-card.js:79]: Right, update in [73af2dd](https://github.com/mozilla/blurts-server/pull/2960/commits/73af2dd1248ca2175361c5fed4ac561c3c1092d1). We adjusted the CSP to this: https://github.com/mozilla/blurts-server/pull/2902/files.

[the remark this answers is not in the document]
Comment (Vinnl) [reply src/views/emails/email-breach-card.js:79]: Sounds good!

File: src/utils/breach-logo.js (hunk 2 above)
Comment (flozia) [reply src/utils/breach-logo.js]: I like both options. You are right though that another class does not hurt anyone and will reduce unnecessary complexity: [fd0a7d8](https://github.com/mozilla/blurts-server/pull/2960/commits/fd0a7d8b878eae5e438ceaa61924ff3e4e5e416a).
